xdsl/dialects/printf.py

```diff
@@ -85,9 +87,29 @@ def parse(cls: type[PrintFormatOp], parser: Parser) -> PrintFormatOp:
         return op
 
 
+@irdl_op_definition
+class PrintCharOp(IRDLOperation):
+    """
+    Print a single character
+
+    Equivalent to putchar in C.
+    """
+
+    name = "printf.print_char"
+    char: Operand = operand_def()
```

Comment:
I think you should restrict it to `i32` here?

Reply:
I've currently refactored this to be constrained on i8's as suggested by @superlopuh